aztec/src/main/kotlin/org/wordpress/aztec/watchers/DeleteMediaElementWatcher.kt

```diff
@@ -0,0 +1,46 @@
+package org.wordpress.aztec.watchers
+
+import android.text.Editable
+import android.text.TextWatcher
+import org.wordpress.aztec.AztecText
+import org.wordpress.aztec.Constants
+import org.wordpress.aztec.spans.AztecMediaSpan
+import java.lang.ref.WeakReference
+
+class DeleteMediaElementWatcher(aztecText: AztecText) : TextWatcher {
+
+    private val aztecTextRef: WeakReference<AztecText?> = WeakReference(aztecText)
+
+    private var deletedMedia: Boolean = false
+    private var mediaSpan: AztecMediaSpan? = null
+
+    override fun beforeTextChanged(text: CharSequence, start: Int, count: Int, after: Int) {
+        deletedMedia = count > 0 && text[start + count - 1] == Constants.IMG_CHAR
+
+        if (deletedMedia) {
+            val aztecText = aztecTextRef.get()
+            mediaSpan = aztecText?.text?.getSpans(start, start + count, AztecMediaSpan::class.java)?.firstOrNull()
+        }
+    }
+
+    override fun onTextChanged(text: CharSequence, start: Int, before: Int, count: Int) {
+        // no op
+    }
+
+    override fun afterTextChanged(text: Editable) {
+        if (aztecTextRef.get()?.isTextChangedListenerDisabled() ?: true) {
+            return
+        }
+
+        if (deletedMedia) {
```

Comment:
Another thing you'll notice is that the listener isn't called when deleting audio media. The problem is that you save the `deletedMedia` flag in `beforeTextChanged()`. However, another text manipulation is triggered (some parser processing) before the `afterTextChanged()` gets called and it changes your `deletedMedia` flag value to `false`. The solution is would be not to store any instance variables in the watcher class but rather save the information somewhere else (maybe keep a list of deleted spans in `AztecText`). I know some of the other watchers have the same issue, I'll need to go back and fix them.